docs/headerparser_usage.md

```diff
@@ -0,0 +1,67 @@
+## Headerparser
+
+headerparser is a tool that assists in writing device system call descriptions for Syzkaller.
+
+In order to make Syzkaller smarter when it comes to fuzzing a device node, you can provide it with
+information about the ioctl argument struct types it expects.
+
+However, in certain cases the number of argument struct types might be high, increasing the amount of manual
+effort that goes into writing the description files for the struct types.
+
+In order to ease the effort of writing ioctl argument type description files, headerlib does a best-effort job at generating them for you. You will still need to manually select the appropriate Syzkaller data type from the list of types [here](https://github.com/google/syzkaller/blob/master/docs/syscall_descriptions_syntax.md).
+
+## Dependencies
+Headerlib uses pycparser. You can install pycparser using pip.
+
+```shell
+$ pip install pycparser
+```
+
+## Using headerparser
+```shell
+$ python headerparser.py --device th_b --filenames=./test_headers/th_b.h
+B {
+                            B1   len|fileoff|flags|intN
+                            B2   len|fileoff|flags|intN
+}
+struct_containing_union {
+                     something   len|fileoff|flags|intN
+                 a_uion.a_char   string|filename
+                  a_uion.B_ptr   ptr|buffer|array
+}
+```
+
+You can copy paste the content underneath the `Structure Metadata` over to your Syzkaller device description.
+
+## Something breaks
+Let us try parsing `test_headers/th_a.h` header file to generate argument structs.
+
+```shell
+$ python headerparser.py --device th_a --filenames=./test_headers/th_a.h
+	...
+	<snip>
```

Comment:
I meant to make the tool print just 1 line. Printing 30 lines for 1 error is too much.

Reply:
ah I see, yes, ill make this change